### ISSUE TITLE
Armbian : Window manager won't start on clean installs

### ISSUE THREAD

**Issue by matt335672** [neutrinolabs/xrdp#1672]
**Edit:** At first I thought this affected Raspbian too, but it only affects Armbian.

At present there's a conflict between the `startwm.sh` provided for Debian in `/etc/xrdp/startwm.sh` and the file `/etc/profile.d/armbian-check-first-login.sh` which is present on most Armbian systems.

The files in `/etc/profile.d` on Debian/Ubuntu are scanned by all Bourne shells. Since this includes the Bourne shell `/bin/sh`, these files need to be compatible with whatever `/bin/sh` is linked to (normally dash).

 `/etc/profile.d/armbian-check-first-login.sh` contains bash-isms which dash doesn't like in `sh` mode. Since the first line of `/etc/xrdp/startwm.sh` is `#!/bin/sh` this can cause the window manager startup to fail.

To see if you are affected by this issue, enter this command on the command line:-
```
/bin/sh -x /etc/profile
```

If the output terminates with the following line you have this problem:-

```
/etc/profile.d/armbian-check-first-login.sh: 20: Syntax error: "(" unexpected
```

The best current workaround is to edit `/etc/xrdp/startwm.sh` and replace the first line with `#!/bin/bash`

I'll raise this issue with the Armbian team and see where it goes.

See also #1630

**Comment by matt335672** [comment]
Issue armbian/build#2185 created on the Armbian project

**Comment by matt335672** [comment]
This has now been addressed in the Armbian repo via PR armbian/build#2190, and so should be available in Armbian builds 20.08.2 and later.

If you're running an earlier Armbian build, use one of the workarounds mentioned above.
